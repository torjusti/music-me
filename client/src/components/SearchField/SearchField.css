input {
  width: 80%;
  font-size: 1.5rem;
  padding: 0 1rem 0 1rem;
  border-radius: 2rem;
  outline: 0;
}
