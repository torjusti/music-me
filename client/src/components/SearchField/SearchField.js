import React, { Component } from 'react';
import styles from './SearchField.module.css';
importÂ { Input, Icon, Button } from 'semantic-ui-react';

export default class SearchField extends Component {
  state = {
    searchValue: '',
  };

  handleChange = input => {
    this.setState({
      searchValue: input,
    });
  };

  render() {
    return (
      <div>
        <Input
          icon="search"
          placeholder="Search ..."
          className={styles.searchField}
        />
        <Button
          onClick={this.handleChange}>
          Search
          <Icon name="angle right"/>
        </Button>
      </div>
    );
  }
}
