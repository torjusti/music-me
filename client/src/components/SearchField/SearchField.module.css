.container {
  display: flex;
}

.searchField {
  width: 100%;
}

.searchContainer {
  flex: 1;
}

.searchButton {
  margin-left: 1rem;
  display: inline-block;
}
