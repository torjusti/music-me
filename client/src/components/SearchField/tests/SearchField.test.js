import React from 'react';
import { shallow } from 'enzyme';
import SearchField from '../SearchField';
import configureMockStore from 'redux-mock-store'

describe('<SearchField />', () => {

  // Create the mock store
  const mockStore = configureMockStore();

  let wrapper, store;

  beforeEach(() => {
    store = mockStore();
    // Shallow render the container passing in the mock store
    wrapper = shallow(
      <SearchField store={store} />
    );
  });

  it('should render SearchField snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render SearchField correctly', () => {
    expect(wrapper.length).toEqual(1);
  });
});
