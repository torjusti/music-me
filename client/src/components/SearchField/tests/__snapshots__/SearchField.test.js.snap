// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchField
    clearQuery={[Function]}
    setQuery={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="searchContainer"
        >
          <Input
            className="searchField"
            icon="search"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Type a value to search for"
            type="text"
            value=""
          />
        </div>,
        <div
          className="searchButton"
        >
          <Button
            as="button"
            onClick={[Function]}
            role="button"
          >
            Search
            <Icon
              as="i"
              name="angle right"
            />
          </Button>
        </div>,
      ],
      "className": "container",
      "data-component": "SearchField",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Input
            className="searchField"
            icon="search"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Type a value to search for"
            type="text"
            value=""
          />,
          "className": "searchContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "searchField",
            "icon": "search",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Type a value to search for",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            as="button"
            onClick={[Function]}
            role="button"
          >
            Search
            <Icon
              as="i"
              name="angle right"
            />
          </Button>,
          "className": "searchButton",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": Array [
              "Search",
              <Icon
                as="i"
                name="angle right"
              />,
            ],
            "onClick": [Function],
            "role": "button",
          },
          "ref": null,
          "rendered": Array [
            "Search",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "name": "angle right",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="searchContainer"
          >
            <Input
              className="searchField"
              icon="search"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Type a value to search for"
              type="text"
              value=""
            />
          </div>,
          <div
            className="searchButton"
          >
            <Button
              as="button"
              onClick={[Function]}
              role="button"
            >
              Search
              <Icon
                as="i"
                name="angle right"
              />
            </Button>
          </div>,
        ],
        "className": "container",
        "data-component": "SearchField",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Input
              className="searchField"
              icon="search"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Type a value to search for"
              type="text"
              value=""
            />,
            "className": "searchContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "searchField",
              "icon": "search",
              "onChange": [Function],
              "onKeyPress": [Function],
              "placeholder": "Type a value to search for",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              as="button"
              onClick={[Function]}
              role="button"
            >
              Search
              <Icon
                as="i"
                name="angle right"
              />
            </Button>,
            "className": "searchButton",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": Array [
                "Search",
                <Icon
                  as="i"
                  name="angle right"
                />,
              ],
              "onClick": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": Array [
              "Search",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "angle right",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
